autoload/test/java/maven.vim

```diff
@@ -0,0 +1,39 @@
+if !exists('g:test#java#maven#file_pattern')
+  let g:test#java#maven#file_pattern = '\v^.*[Tt]est\.java$'
+endif
+
+function! test#java#maven#test_file(file) abort
+  return a:file =~? g:test#java#maven#file_pattern
+endfunction
+
+function! test#java#maven#build_position(type, position) abort
+  if a:type == 'nearest'
+    let name = s:nearest_test(a:position)
+    if !empty(name)
+      return [name]
+    else
+      return [filename]
+    endif
+  elseif a:type == 'file'
+    let strip_extension = split(a:position['file'], "\.java")[1]
```

Comment:
Here, we have a file like `src/test/java/main/MyTest.java`. I need to reduce this to `MyTest`. I'm not sure if there is a better way to do this with vimscript.